### PR TITLE
support for SELECT special functions

### PR DESCRIPTION
**Pull request by kanha-gupta**
Ref #24200 .

Changes proposed in this pull request:
  -

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by kanha-gupta** [comment]
@strongduanmu Hello, I request you to review it please.
task #1

**Comment by strongduanmu** [comment]
Hi @kanha-gupta, can you refer issue #24200 for this pr?

**Comment by kanha-gupta** [comment]
@strongduanmu Hello
The code is getting build error Failures: 
[ERROR]   InternalMySQLParserIT>InternalSQLParserIT.assertSupportedSQL:60 
SQL Case ID : select_char
SQL         : SELECT CHAR(77,121,83,81,'76')
`ProjectionsSegment`'s stop index assertion error: 

Expected: is <44>
     but: was <29>

XML code already have expected value at stop index & therefore I couldnt reach to solution.
Can you please suggest error resolution ? 
Thank you 
(select_char test is being passed successfully in SQLNodeConverterEngineIT.java)

File: test/it/parser/src/main/resources/case/dml/select-special-function.xml [hunk 1]
```diff
@@ -125,10 +125,26 @@
         </projections>
     </select>
     <select sql-case-id="select_char">
-        <projections start-index="7" stop-index="29">
-            <expression-projection text="CHAR(77,121,83,81,'76')" start-index="7" stop-index="29">
+        <projections start-index="7" stop-index="44">
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/case/dml/select-special-function.xml:128]: Why modify `stop-index` to `44`? The projection stop index in sql `SELECT CHAR(77,121,83,81,'76')` is 29, you should keep it with correct value.

File: test/it/parser/src/main/resources/case/dml/select-special-function.xml [hunk 2]
```diff
@@ -125,10 +125,26 @@
...
+                            <literal-expression value="121" start-index="15" stop-index="17" />
+                        </parameter>
+                        <parameter>
+                            <literal-expression value="83" start-index="20" stop-index="21" />
+                        </parameter>
+                        <parameter>
+                            <literal-expression value="81" start-index="24" stop-index="25" />
+                        </parameter>
+                        <parameter>
+                            <literal-expression value="'76'" start-index="29" stop-index="32" />
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/case/dml/select-special-function.xml:145]: This start-index and stop-index seems also wrong.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: test/it/parser/src/main/resources/case/dml/select-special-function.xml (hunk 1 above)
Comment (kanha-gupta) [reply test/it/parser/src/main/resources/case/dml/select-special-function.xml]: I optimised it. Thanks alot :)

**Comment by kanha-gupta** [comment]
@strongduanmu Hello,
Can you please suggest How to parse Identifiers & symbols in MySQLStatementSQLVisitor.java & related XML files from their Antlr grammar files ? 
I tried finding a solution but I cant come up to solution looking at existing codebase.
Thank you

**Comment by strongduanmu** [comment]
Hi @kanha-gupta, can you merge master branch?

**Comment by kanha-gupta** [comment]
@strongduanmu Done, please check.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Review by strongduanmu — APPROVED** [review]
Good job, merged.

**Pull request merged**